File: src/main/java/com/ada/conversation/infrastructure/in/rest/filter/NormalizeChatRequestFilter.java
package com.ada.conversation.infrastructure.in.rest.filter;

import com.ada.conversation.application.dto.ChatRequest; import com.ada.conversation.application.port.in.RequestFilter; import org.springframework.core.annotation.Order; import org.springframework.stereotype.Component;
@Component @Order(10) public class NormalizeChatRequestFilter implements RequestFilter { public boolean supports(ChatRequest request){return true;} public ChatRequest apply(ChatRequest request){return new ChatRequest(request.message().trim(),request.requestedModel());} }

File: src/main/kotlin/com/ada/conversation/infrastructure/in/rest/filter/NormalizeChatRequestFilter.kt
package com.ada.conversation.infrastructure.`in`.rest.filter

import com.ada.dto.ChatRequest
import com.ada.conversation.application.port.`in`.RequestFilter
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component

@Component
@Order(10)
class NormalizeChatRequestFilter : RequestFilter {
    override fun supports(request: ChatRequest): Boolean = true

    override fun apply(request: ChatRequest): ChatRequest = request.copy(message = request.message.trim())
}
